### ISSUE TITLE
Drag and drop from SMB file explorer (network driver) produces incorrect file paths with WSL

### ISSUE THREAD

**Issue by fengyichui** [mintty/mintty#1130]
Drag network driver file (SMB) into terminal window with WSL should produce the text:
> /mnt/net/5p11/projects

instead, it produces:
> //192.168.5.11/file_server/projects

My mount is:
> \\192.168.5.11\file_server on /mnt/net/5p11 type drvfs (rw,relatime,case=off)

Draging local file (C:\ D:\) into terminal can produce correct paths.

**Comment by mintty** [comment]
How is the mount point established? Windows mount plus WSL /etc/fstab?
Please provide complete instructions to reproduce the issue.

**Comment by fengyichui** [comment]
I use WSL /etc/fstab to mount, show as follow:
> LABEL=cloudimg-rootfs	/	 ext4	defaults	0 0
\\192.168.5.10\huntersun            /mnt/net/5p10   drvfs  defaults  0  0
\\192.168.5.11\file_huntersun       /mnt/net/5p11   drvfs  defaults  0  0

**Comment by mintty** [comment]
Mounting by IP does not work for me. If I mount by hostname, drag-and-drop works.

**Comment by fengyichui** [comment]
My company‘s files server may have no hostname, and may only be accessed through IP.
I use `findsmb` or `nslookup` to search the hostname, but nothing.

**Comment by mintty** [comment]
I don't know why the mount via IP did not work for me. Did you configure anything else?
You could try to assign the respective host a local hostname in your /etc/hosts file and then mount by name.

**Comment by fengyichui** [comment]
I don't configure else with mounting.

I have tried this, and I also modified win10's hosts file. It does not work.

**Comment by mintty** [comment]
> If I mount by hostname, drag-and-drop works.

Actually, only the fallback-mapping seems to work, which assings drive-letter-X mounts to the default /mnt/X mount point. To be checked.

**Comment by fengyichui** [comment]
Yes, it works . I mapped a network drive in windows explorer, drag-and-drop works.

Is the path conversion implemented by windows itself, not by mintty?

**Comment by mintty** [comment]
Path conversion is implemented by mintty, but the path information coming from Explorer will be different after driver letter mapping. Also, I guess this may only work if you mount to the standard /mnt/X. What if you mount to /mnt/othername or /tmp/othermount?

**Comment by fengyichui** [comment]
I mount to /mnt/othername. Because `/mnt/X` is not easy to remember, so I give a friendly name.

I often don’t map network file drives in Windows Explorer, and just mount in WSL.

**Comment by mintty** [comment]
How can you drag (and drop) from Explorer if the share is not mounted there?
Or is it something else that you drag from? In that case, mintty is likely not prepared to handle it.

**Comment by fengyichui** [comment]
I enter the path directly in the address bar of the explorer, the path like that:
> \\192.168.3.1\Western_Digital_usb1_1\Config

The screenshot show as follow：
![1](https://user-images.githubusercontent.com/17896291/141664011-cf47f8e7-ef4a-4084-9fe7-c1b16002a5c3.png)

**Comment by Biswa96** [comment]
The 192.168.x.x <-> /mnt/net translation happen inside WSL environment, mintty program is not aware of that. One thing can be done but I have _no_ idea how to do it. wslbridge2-backend can read/parse that `/etc/fstab` and report the results back to wslbridge2.exe or mintty using the controlSock[^1] socket. Also don't know if it gets confused with automount.mountFsTab[^2] field in /etc/wsl.conf file.

[^1]: https://github.com/Biswa96/wslbridge2/blob/master/src/wslbridge2-backend.cpp
[^2]: https://docs.microsoft.com/en-us/windows/wsl/wsl-config

**Comment by mintty** [comment]
Actually mintty considers the WSL fstab already, reading it via its converted Windows path.
Currently, however, it interprets and maps its contents only partially.

Also that would not help for dynamic mounts (using mount command, without fstab entry) but I understand that's not your use case anyway, also the mounts would not appear in the Windows Explorer.

**Comment by mintty** [comment]
I've uploaded a tweak to consider extended /etc/fstab entries like
//host/share /mnt/mount/dir
//host/share/subdir /mnt/mount/dir

Can you test whether this covers your case?

**Comment by fengyichui** [comment]
It seems not work in my case. I compile it with Cygwin. How can I help to debug it ?

**Comment by mintty** [comment]
OK, then please describe your case, again and precisely. How do you mount, what's the contents of /etc/fstab and /etc/wsl.conf, what's your mount on Windows side, what does Explorer display when you just click on the address bar.

**Comment by fengyichui** [comment]
I mount with /etc/fstab:
``` fstab
LABEL=cloudimg-rootfs	/	 ext4	defaults	0 0
//192.168.5.10/huntersun            /mnt/net/5p10   drvfs  defaults  0  0
//192.168.5.11/file_huntersun       /mnt/net/5p11   drvfs  defaults  0  0
```

/etc/wsl.conf:
``` ini
[automount]
enabled = true
options = "metadata,umask=22,fmask=11"
[user]
default=myname
[network]
generateHosts = false
```

I do not mount on Windows side, just use the SMB IP address.

When I drag, the windows explorer display on address bar:
> \\\\192.168.5.11\file_huntersun\users\liq

After drop, the Mintty show:
> //192.168.5.11/file_huntersun/users/liq/c.bin

**Comment by mintty** [comment]
Implemented handling of UNC paths, please test.
Didn't check with automount enabled, however, hope it works.

**Comment by fengyichui** [comment]
It works!  👍👍👍

**Comment by mintty** [comment]
Released 3.5.3.
